plugins/enthusiast-common/enthusiast_common/injectors/base.py

```diff
@@ -0,0 +1,47 @@
+from abc import ABC, abstractmethod
+
+from .document import BaseDocumentRetriever
+from .product import BaseProductRetriever
+
+
+class BaseInjector(ABC):
+    @property
+    @abstractmethod
+    def document_retriever(self) -> BaseDocumentRetriever:
+        pass
+
+    @property
+    @abstractmethod
+    def product_retriever(self) -> BaseProductRetriever:
+        pass
+
+    @document_retriever.setter
+    @abstractmethod
+    def document_retriever(self, retriever: BaseDocumentRetriever) -> None:
+        pass
+
+    @product_retriever.setter
+    @abstractmethod
+    def product_retriever(self, retriever: BaseProductRetriever) -> None:
+        pass
```

Comment:
I'm not sure if we want to have setters in this interface - user's implementation of agents probably shouldn't mess with the retrievers (?)

Reply:
I was thinking of additional retriever that developer may want to have available inside a tool